storage/connectors/redis/src/main/java/feast/storage/connectors/redis/retriever/FeatureRowDecoder.java

```diff
@@ -36,60 +41,100 @@ public FeatureRowDecoder(String featureSetRef, FeatureSetSpec spec) {
   }
 
   /**
-   * A feature row is considered encoded if the feature set and field names are not set. This method
-   * is required for backward compatibility purposes, to allow Feast serving to continue serving non
-   * encoded Feature Row ingested by an older version of Feast.
+   * Check if encoded feature row v1 is encoded. The Feature Row v1 encoding defines that a Feature
+   * Row is is considered encoded if both it's feature set reference and fields names are not set .
+   * The no. of fields in the feature row should also match up with the number of fields in the
+   * Feature Set spec. NOTE: This method is deprecated and will be removed in Feast v0.7.
    *
    * @param featureRow Feature row
    * @return boolean
    */
-  public boolean isEncoded(FeatureRow featureRow) {
+  @Deprecated
+  private boolean isEncodedV1(FeatureRow featureRow) {
     return featureRow.getFeatureSet().isEmpty()
-        && featureRow.getFieldsList().stream().allMatch(field -> field.getName().isEmpty());
+        && featureRow.getFieldsList().stream().allMatch(field -> field.getName().isEmpty())
+        && featureRow.getFieldsList().size() == spec.getFeaturesList().size();
   }
 
   /**
-   * Validates if an encoded feature row can be decoded without exception.
+   * Check if encoded feature row v2 is encoded. The Feature Row v2 encoding defines that a Feature
+   * Row is f is considered encoded if it's both it feature set reference and fields names are set.
    *
    * @param featureRow Feature row
    * @return boolean
    */
-  public boolean isEncodingValid(FeatureRow featureRow) {
-    return featureRow.getFieldsList().size() == spec.getFeaturesList().size();
+  private boolean isEncodedV2(FeatureRow featureRow) {
+    return featureRow.getFieldsList().stream().allMatch(field -> !field.getName().isEmpty());
```

Comment:
From a performance perspective, would it make sense to rather use `anyMatch()` or test the first field?

Reply:
Updated

Reply:
Changed to `anyMatch()`